scripts/plugins/anchor-markdown-headings.js

```diff
@@ -1,14 +1,42 @@
 'use strict'
 
+// Check whether we've got the comment
+const ANCHOR_COMMENTREG = /<!--[\w-]+-->/gi
+
+/**
+ * This module will automatically replace the header style words
+ * (such as `#`,`##`...ect) to an anchor name.
+ *
+ * This processor will make all the English characters (26 letters)
+ * into their lower case, for others in the middle of words,
+ * they will become `-` in default.
+ *
+ * If your title has some non-English characters, please use
+ * `<!---->` to quote your English anchor name inside, with your
+ * own title beside it.
+ */
 module.exports = function anchorMarkdownHeadings (text, level, raw) {
-  const escapedText = raw
-    .replace(/(\[([^\]]+)]\([^)]+\))/g, '$2')
-    .replace(/[^\w]+/g, '-')
-    .replace(/-{2,}/g, '-')
-    .replace(/(^-|-$)/g, '')
-    .toLowerCase()
-  return '<h' + level + ' id="header-' + escapedText + '">' + text + '<a name="' +
-    escapedText + '" class="anchor" href="#' +
-    escapedText + '" aria-labelledby="header-' +
-    escapedText + '"></a></h' + level + '>'
+  let anchorTitle = null
+
+  // If we've checked the title has a comment symbol,
+  // we reguard it as the forcely-assigned anchor link name
+  // for titles with non-English characters
+  const anchorTitleArray = ANCHOR_COMMENTREG.exec(raw)
+  if (anchorTitleArray !== null) {
+    anchorTitle = anchorTitleArray[0].replace('<!--', '').replace('-->', '')
```

Comment:
If we do the matching directly in the RegExp above, we can use `anchorTitle = anchorTitleArray[1]`.